driver/pace/driver/diagnostics.py

```diff
@@ -174,13 +172,13 @@ def _compute_column_integral(
         q_in: tracer mixing ratio
         delp: pressure thickness of atmospheric layer
     """
-    if len(q_in.shape) < 3:
-        assert ShapeError(f"{name} does not have vertical levels.")
+    if q_in.dims[2] != pace.util.Z_DIM:
+        raise NotImplementedError(
+            "this function assumes the z-dimension is the third dimension"
+        )
```

Comment:
Does `q_in` always have 3 dims so that indexing `[2]` directly does not raise an `IndexError`? Maybe make this
```python
assert len(q_in.dims) > 2
if q_in.dims[2] != pace.util.Z_DIM:
```

but maybe that's already assumed throughout the code.

Reply:
updated